amazonec2/amazonec2_endpoint.bal

```diff
@@ -340,27 +368,34 @@ public remote function Client.describeImages(string... imgIdArr) returns Image[]
     canonicalQueryString = canonicalQueryString + "Version" + "=" + API_VERSION;
     string constructCanonicalString = "/?" + canonicalQueryString;
     request.setHeader(HOST, host);
-    generateSignature(request, self.accessKeyId, self.secretAccessKey, self.region, GET, requestURI, "",
+    var signature = generateSignature(request, self.accessKeyId, self.secretAccessKey, self.region, GET, requestURI, "",
         canonicalQueryString);
-    var response = self.amazonClient->get(untaint constructCanonicalString, message = request);
-
-    if (response is http:Response) {
-        int statusCode = response.statusCode;
-        var amazonResponse = response.getXmlPayload();
-        if (amazonResponse is xml) {
-            if (statusCode == 200) {
-                return getSpawnedImageList(amazonResponse);
+
+    if (signature is error) {
+        error err = error(AMAZONEC2_ERROR_CODE, { ^"error": signature.detail(),
+            message: "Error occurred while generating the amazon signature header" });
+        return err;
+    } else {
+        var response = self.amazonClient->get(untaint constructCanonicalString, message = request);
+
+        if (response is http:Response) {
+            int statusCode = response.statusCode;
+            var amazonResponse = response.getXmlPayload();
+            if (amazonResponse is xml) {
+                if (statusCode == http:OK_200) {
```

Comment:
shall we just check 
```suggestion
                if (response.statusCode == http:OK_200) {
```
instead of assigning it to a variable since this is the only usage.

Please check all places.